packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -79,18 +79,26 @@ function useInsertionPoint( {
 				getBlockRootClientId,
 				getBlockIndex,
 				getBlockOrder,
-			} = select( blockEditorStore );
+				getInserterInsertionPoint,
+			} = unlock( select( blockEditorStore ) );
 			const selectedBlockClientId = getSelectedBlockClientId();
 
 			let _destinationRootClientId = rootClientId;
 			let _destinationIndex;
+			const insertionPoint = getInserterInsertionPoint();
 
 			if ( insertionIndex !== undefined ) {
 				// Insert into a specific index.
 				_destinationIndex = insertionIndex;
 			} else if ( clientId ) {
 				// Insert after a specific client ID.
 				_destinationIndex = getBlockIndex( clientId );
+			} else if (
+				insertionPoint?.insertionIndex &&
+				insertionPoint?.rootClientId
+			) {
+				_destinationRootClientId = insertionPoint.rootClientId;
+				_destinationIndex = insertionPoint.insertionIndex;
```

Comment:
Putting this after the `insertionIndex` or `clientId ` to preserve the existing API. This lets anyone who passes these values in via the props to still have full control.